packages/react/src/components/Table/Table.story.helpers.jsx

```diff
@@ -0,0 +1,902 @@
+      rules: [
+        {
+          id: 'rsiru4rjba',
+          columnId: 'date',
+          operand: 'CONTAINS',
+          value: '19',
+        },
+        {
+          id: '34bvyub9jq',
+          columnId: 'boolean',
+          operand: 'EQ',
+          value: 'true',
+        },
+      ],
+    },
+    filterColumns: getTableColumns(),
+  },
+  {
+    filterId: 'next-filter',
+    /** Text for main tilte of page */
+    filterTitleText: 'select=Option c, boolean=false',
+    /** Text for metadata for the filter */
+    filterMetaText: `last updated: 2021-03-11 15:34:01`,
+    /** tags associated with particular filter */
+    filterTags: ['fav', 'other-tag'],
+    /** users that have access to particular filter */
+    filterAccess: [
+      {
+        username: 'Example-User',
+        email: 'example@pal.com',
+        name: 'Example User',
+        access: 'edit',
+      },
+      {
+        username: 'Other-User',
+        email: 'other@pal.com',
+        name: 'Other User',
+        access: 'read',
+      },
+    ],
+    /** All possible users that can be granted access */
+    filterUsers: [
+      {
+        id: 'teams',
+        name: 'Teams',
+        groups: [
+          {
+            id: 'team-a',
+            name: 'Team A',
+            users: [
+              {
+                username: '@tpeck',
+                email: 'tpeck@pal.com',
+                name: 'Templeton Peck',
+              },
+              {
+                username: '@jsmith',
+                email: 'jsmith@pal.com',
+                name: 'John Smith',
+              },
+            ],
+          },
+        ],
+      },
+      {
+        username: 'Example-User',
+        email: 'example@pal.com',
+        name: 'Example User',
+      },
+      {
+        username: 'Test-User',
+        email: 'test@pal.com',
+        name: 'Test User',
+      },
+      {
+        username: 'Other-User',
+        email: 'other@pal.com',
+        name: 'Other User',
+      },
+    ],
+    /**
+     * the rules passed into the component. The RuleBuilder is a controlled component, so
+     * this works the same as passing defaultValue to a controlled input component.
+     */
+    filterRules: {
+      id: '14p5ho3pcu',
+      groupLogic: 'ALL',
+      rules: [
+        {
+          id: 'rsiru4rjba',
+          columnId: 'select',
+          operand: 'EQ',
+          value: 'option-C',
+        },
+        {
+          id: '34bvyub9jq',
+          columnId: 'boolean',
+          operand: 'EQ',
+          value: 'false',
+        },
+      ],
+    },
+    filterColumns: getTableColumns(),
+  },
+];
+
+export const getTableKnobs = (enableAllKnobs) => ({
+  selectedTableType: select('Type of Table', ['Table', 'StatefulTable'], 'StatefulTable'),
+  tableMaxWidth: select('Demo table max-width', ['300px', '600px', '900px', 'none'], 'none'),
+
+  // Header
+  secondaryTitle: text('Title shown in bar above header row (secondaryTitle)', 'Table playground'),
```

Comment:
Just a thought, should we use the third parameter on the knobs to actually group them in storybook tabs in addition to grouping them together? https://github.com/storybookjs/addon-knobs Is that even compatible with the new controls add-on? Just wondering, because there are _a lot_ of knobs in there...It definitely helps that they're grouped by function though!

Reply:
I really like that idea. Unfortunately that is not compatible with the new controls add-on, but there is a feature request to add groups. I'll add the groups here anyway, it is not a big effort to remove them if needed for the upgrade to the controls add-on.